homeassistant/components/signal_messenger/notify.py

```diff
@@ -43,37 +55,93 @@ def get_service(hass, config, discovery_info=None):
 class SignalNotificationService(BaseNotificationService):
     """Implement the notification service for SignalMessenger."""
 
-    def __init__(self, recp_nrs, signal_cli_rest_api):
+    def __init__(
+        self,
+        recp_nrs: list[str],
+        signal_cli_rest_api: SignalCliRestApi,
+    ) -> None:
         """Initialize the service."""
 
         self._recp_nrs = recp_nrs
         self._signal_cli_rest_api = signal_cli_rest_api
 
-    def send_message(self, message="", **kwargs):
-        """Send a message to a one or more recipients.
-
-        Additionally a file can be attached.
-        """
+    def send_message(self, message: str = "", **kwargs: Any) -> None:
+        """Send a message to a one or more recipients. Additionally a file can be attached."""
 
         _LOGGER.debug("Sending signal message")
 
         data = kwargs.get(ATTR_DATA)
 
-        filenames = None
-        if data is not None:
-            if ATTR_FILENAMES in data:
-                filenames = data[ATTR_FILENAMES]
-            if ATTR_FILENAME in data:
-                _LOGGER.warning(
-                    "The 'attachment' option is deprecated, please replace it with 'attachments'. This option will become invalid in version 0.108"
-                )
-                if filenames is None:
-                    filenames = [data[ATTR_FILENAME]]
-                else:
-                    filenames.append(data[ATTR_FILENAME])
+        filenames = self.get_filenames(data)
+        attachments_as_bytes = self.get_attachments_as_bytes(
+            data, CONF_MAX_ALLOWED_DOWNLOAD_SIZE_BYTES
+        )
 
         try:
-            self._signal_cli_rest_api.send_message(message, self._recp_nrs, filenames)
+            self._signal_cli_rest_api.send_message(
+                message, self._recp_nrs, filenames, attachments_as_bytes
+            )
         except SignalCliRestApiError as ex:
             _LOGGER.error("%s", ex)
             raise ex
+
+    @staticmethod
+    def get_filenames(data: Any) -> list[str] | None:
+        """Extract attachment filenames from data."""
+        if data is None:
+            return None
+
+        if ATTR_FILENAMES in data:
+            if isinstance(data[ATTR_FILENAMES], list):
```

Comment:
Validate date using voluptuous?

See here for an example: https://github.com/home-assistant/core/blob/56c577c8321d543197c469eceaaefe660ed2b698/homeassistant/components/slack/notify.py#L44

Reply:
Thanks - now added in 9d6d7b1